### ISSUE TITLE
Documentation Improvement: nuxtApp, useNuxtApp, composables, plugins

### ISSUE THREAD

**Issue by jshimkoski** [nuxt/framework#6719]
### Describe the feature

It would be nice if the documentation had answers to these questions:

* Can you use a plugin in a composable?
* Can you use a plugin within another plugin?
* Can you use a composable in a plugin?
* Can you use a composable within another composable?
* And most importantly, what does the Nuxt 3 lifecycle look like and when do different pieces of an app become available to the `nuxtApp` context?

Back story: While porting a massive project from Nuxt 2 -> Nuxt 3, my app exploded after RC.7 due to the usage of a plugin within another plugin (oddly, my app worked without issue up to that point). I had to combine the plugins to bring the app back to life. This took hours to troubleshoot and it was a pretty easy thing to mess up.

What I expected:
"I'm providing this plugin to my app, I should be able to access it anywhere the `nuxtApp` context is available."

What really happens:
"I'm providing this plugin to my app, I can access it in my layouts, pages and components, but I think that's about it."

So, long story short, it would be awesome if the documentation could be updated to answer those 5 questions.

I could help contribute an update to the documentation if provided the answers by the peeps in the know.

Thanks!

### Additional information

- [X] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://v3.nuxtjs.org/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/framework/discussions) and [issues](https://github.com/nuxt/framework/issues).

**Comment by 516310460** [comment]
![image](https://user-images.githubusercontent.com/17541676/185290988-0e535712-ad67-461b-a88d-995cef20d07b.png)

**Comment by 516310460** [comment]
1/ Can you use a plugin in a composable?
```ts
const cxt = useNuxtApp()
```
2/ Can you use a plugin within another plugin?
```ts
export default defineNuxtPlugin((NuxtApp) => {
  console.log(NuxtApp)
 // ******
return { test: 'test demo' }
})
```
3/ Can you use a composable in a plugin? And 4/ Can you use a composable within another composable?
```ts
const foo = useFoo()
console.log(foo)
```

**Comment by 516310460** [comment]
5/
Link: https://github.com/nuxt/framework/discussions/6734

**Comment by 516310460** [comment]
> I've upgraded Nuxt2 to Nuxt3 and the overall story with composable is frustrating :( When application becomes large there is a need to call composables from all over the app, since after refactoring most logic is extracted into sepatate files and context is lost. While few apis do take nuxtApp as a parameter (like useRequestEvent) others don't (like nagivateTo) which makes huge inconvenience. If there is a decision that nuxtApp should be passed explicit as much as possible, then all api's should have it as explicit argument

You can understand the context of 'usenuxtapp' in 'nuxt2'

**Comment by Lanayx** [comment]
> Pugins can be injected in `useNuxtApp`
I'm not talking about plugins, just helper logic files, say on some page you make request, then await in another file, then in some another file navigateTo or useRouter and it will fail with missing context and even though you could pass nuxtApp explicitly this doesn't help, since most api's don't take it as a parameter

**Comment by iPrytz** [comment]
I'm trying to use navigateTo inside a composable, its not working because missing nuxt context. Will that be possible?

**Comment by 516310460** [comment]
> I'm trying to use navigateTo inside a composable, its not working because missing nuxt context. Will that be possible?

would you please provide a reproduction?

**Comment by iPrytz** [comment]
> > I'm trying to use navigateTo inside a composable, its not working because missing nuxt context. Will that be possible?
> 
> would you please provide a reproduction?

https://codesandbox.io/s/old-frog-muodnd?file=/app.vue

**Comment by danielroe** [comment]
Let's track in https://github.com/nuxt/framework/issues/4104, which is more specific.
